### DISCUSSION TITLE
Exposure filter

### DISCUSSION THREAD

**Discussion by paulm17** [fabricjs/fabric.js#8093] [Q&A] [answered]
Hi Andrea,

I see you assisted with: https://stackoverflow.com/questions/40655994/how-to-implement-exposure-image-filter-using-fabricjs-basefilter

Any chance you could expand on this further?  

I'm not going to be using an external canvas.  But instead Fabric.Image.fromURL.

Something like this:

```
fabric.Image.fromURL(
      useStore.previewBackground.image,
      function (img) {
        img.scaleX =
          (useStore.canvasObj?.current.width * useStore.repositionTemp.value) /
          img.width
        img.scaleY =
          (useStore.canvasObj?.current.height * useStore.repositionTemp.value) /
          img.height

        /*var filter = new fabric.Image.filters.Gamma({
          gamma: [data[0], data[1], data[2]],
        })*/

        console.log(img)

        //img.filters.push(filter)
        //img.applyFilters()

        useStore.canvasObj?.current.setBackgroundImage(img)
        useStore.canvasObj?.current.discardActiveObject()
        useStore.canvasObj?.current.requestRenderAll()
      },
      {
        crossOrigin: "anonymous",
      },
    )
```

I'm stuck how how to calculate the RGB from an "exposure" value from -100 to 100.

Thanks

**Comment by melchiar** [comment, marked as answer]
A value in a range from -100 to 100 can be converted to a value between 0 to 1 like this.

`value = (value + 100) / 200;`

Then just multiply the new value by whatever you decide is your upper gamma limit.

**Reply by paulm17** [reply]
Thanks for the hint.  However, I'm completely lost as I'm looking for an RGB array [red, green, blue] to supply to the gamma func.

http://fabricjs.com/docs/fabric.Image.filters.Gamma.html#gamma

**Reply by melchiar** [reply]
You don't need an rgb array, just use the built in Gamma filter which takes an array of values between 0 and 2.1.

```
var array = [ value * 2.1, value * 2.1, value * 2.1];
var filter = new fabric.Image.filters.Gamma({
  gamma: array
});
object.filters.push(filter);
object.applyFilters()
```

See http://fabricjs.com/docs/fabric.Image.filters.Gamma.html

**Reply by paulm17** [reply]
Awesome, thanks.  This works.  I appreciate your time!

**Comment by paulm17** [comment]
Found a more complete example.  Pretty much, to duplicate the end result of this: 

https://gist.github.com/rijulg/11a2e50804dd5335ca9178522bebfa18
